shanoir-ng-datasets/src/main/java/org/shanoir/ng/dicom/DicomProcessing.java

```diff
@@ -41,4 +47,40 @@ public Attributes getDicomObjectAttributes(DatasetFile image, Boolean isEnhanced
 		}
 	}
 
+    public ExaminationAttributes getDicomExaminationAttributes(Study study, Boolean isEnhanced) throws PacsException {
+		ExaminationAttributes attributes = new ExaminationAttributes();
+		if (study != null) {
+			for (Serie serie : study.getSeries()) {
+				attributes.addAcquisitionAttributes(serie.hashCode(), getDicomAcquisitionAttributes(serie, isEnhanced));
+			}
+		}
+		return attributes;
+    }
+
+	public ExaminationAttributes getDicomExaminationAttributes(Study study) throws PacsException {
+		ExaminationAttributes attributes = new ExaminationAttributes();
+		if (study != null) {
+			for (Serie serie : study.getSeries()) {
+				attributes.addAcquisitionAttributes(serie.hashCode(), getDicomAcquisitionAttributes(serie));
+			}
+		}
+		return attributes;
+    }
+
+	public AcquisitionAttributes getDicomAcquisitionAttributes(Serie serie, Boolean isEnhanced) throws PacsException {
+		AcquisitionAttributes attributes = new AcquisitionAttributes();
+		for (Dataset dataset : serie.getDatasets()) {
+			try {
+				attributes.addDatasetAttributes(dataset.hashCode(), getDicomObjectAttributes(serie.getFirstDatasetFileForCurrentSerie(), isEnhanced));
```

Comment:
same here, you add it by hashCode, and later I think you search it by long id, maybe I am wrong